trade-finance-manager-ui/server/controllers/case/cancellation/validation/validate-previous-page.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Validates the if the URL is one of
+ *  - reason for cancelling page
+ *  - bank request date page
+ *  - effective from date page
+ *  - check details page
+ *
+ * and returns a new relative URL for that page.
+ * Otherwise, returns the deal summary page URL.
+ * @param url - The URL of the previous page, provided in the post request
+ * @param dealId - The deal id
+ * @returns the previous page URL
+ */
+export const validatePreviousPage = (url: string, dealId: string): string => {
```

Comment:
Do we really need to do this? The design has:

_"Are you sure you want to cancel the request?
You'll lose all the data you have entered **and return to deal summary**"_

Instead of this and the following in the controller:

```ts
return res.redirect(validatePreviousPage(req.body.previousPage, _id));
```

Could we not just do the following?

```ts
return res.redirect('/case-summary-url');
```

Reply:
The secondary button `No go back` needs to redirect the user to the previous page.

when the user clicks this button, a POST request is made from (and to) `cancellation/cancel` meaning the `req.headers.referrer` is not the previous page.

To get around this, I'm passing the `previousPage` as a hidden input on the POST request. The downside is that the previous page is now _technically_ a user supplied value and needs to be validated before using it to redirect the user. 

I could get around this by adding the previous page to flash/session but I think this is the easiest solution to implement.

Reply:
Thanks @MarRobSoftwire it's not my call, but this feels like a strange design - visual and therefore technical solution to achieve it.

The `req.body` or DOM could also be manipulated and then there'd be no `previousPage` and the URL would be undefined. This is edge case though.

Can we at least rename `validatePreviousPage` to `getPreviousPageUrl` and remove the `validation` directory please because at the moment It could be read that "previous page" is a validation rule in terms of the form POST.

I get that it's consuming a value from the form body but i'd typically expect anything in a controller's validation directory to be validation rules that are presented to a user.

Reply:
Sure will do,
as to the `previousPage` being undefined - this function will return `/case/${dealId}/deal` if thats the case